langchain/chains/router/base.py

```diff
@@ -0,0 +1,71 @@
+"""Base classes for chain routing."""
+from __future__ import annotations
+
+from abc import ABC
+from typing import Any, Dict, List, Mapping, Optional
+
+from pydantic import Extra
+
+from langchain.callbacks.manager import CallbackManagerForChainRun
+from langchain.chains.base import Chain
+
+
+class RouterChain(Chain, ABC):
+    """Chain that outputs the name of a destination chain and the inputs to it."""
+
+    @property
+    def output_keys(self) -> List[str]:
```

Comment:
could add a method that returns a structured NamedTuple (for typehingint downstream. eg

```
def route(inputs: dict) -> route:
      result = self(inputs)
      return Route(result[destination], result[next_inputs])
```